lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts

```diff
@@ -17,20 +17,52 @@
 
 import { TestBed } from '@angular/core/testing';
 import { MatDialog, MatDialogModule, MatDialogRef } from '@angular/material/dialog';
-import { Observable, Subject } from 'rxjs';
+import { EMPTY, Observable, of, Subject } from 'rxjs';
 import { UnsavedChangesDialogComponent } from './unsaved-changes-dialog.component';
 import { UnsavedChangesGuard } from './unsaved-changes.guard';
+import { AppConfigService, AuthenticationService, AuthGuardService, AuthService, RedirectAuthService } from '@alfresco/adf-core';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { AUTH_CONFIG, DateTimeProvider, OAuthLogger, OAuthService, OAuthStorage, UrlHelperService } from 'angular-oauth2-oidc';
+import { AUTH_MODULE_CONFIG } from '../../auth/oidc/auth-config';
 
 describe('UnsavedChangesGuard', () => {
     let guard: UnsavedChangesGuard;
+    let authService: AuthenticationService;
+    let appConfigService: AppConfigService;
 
     beforeEach(() => {
         TestBed.configureTestingModule({
-            imports: [MatDialogModule]
+            imports: [MatDialogModule, HttpClientTestingModule],
+            providers: [
+                UrlHelperService,
+                DateTimeProvider,
+                { provide: AUTH_CONFIG, useValue: {} },
+                { provide: AUTH_MODULE_CONFIG, useValue: {} },
+                { provide: OAuthService, useValue: {} },
+                { provide: OAuthLogger, useValue: {} },
+                { provide: OAuthStorage, useValue: {} },
+                { provide: AuthService, useValue: {} },
+                { provide: RedirectAuthService, useValue: { onLogin: EMPTY, onTokenReceived: of() } }
+            ]
         });
         guard = TestBed.inject(UnsavedChangesGuard);
+        authService = TestBed.inject(AuthenticationService);
+        TestBed.inject(AuthGuardService);
+        appConfigService = TestBed.inject(AppConfigService);
     });
 
+    const expectGuardToBe = (condition: boolean, done: DoneFn, checkUnsaved?: boolean) => {
+        (guard.canDeactivate() as Observable<boolean>).subscribe((allowed) => {
+            if (checkUnsaved) {
```

Comment:
we can avoid repeating condition with expect and calling done function by doing following: 
```ts
if (checkUnsaved) {
  allowed = guard.unsaved;
}
condition ? expect(allowed).toBeTrue() : expect(allowed).toBeFalse();
done();